chia/cmds/coin_funcs.py

```diff
@@ -105,120 +115,140 @@ def print_coins(
                 break
 
 
-async def async_combine(args: Dict[str, Any], wallet_client: WalletRpcClient, fingerprint: int) -> None:
-    wallet_id: int = args["id"]
-    min_coin_amount = Decimal(args["min_coin_amount"])
-    excluded_amounts = args["excluded_amounts"]
-    number_of_coins = args["number_of_coins"]
-    max_amount = Decimal(args["max_amount"])
-    target_coin_amount = Decimal(args["target_coin_amount"])
-    target_coin_ids: List[bytes32] = [bytes32.from_hexstr(coin_id) for coin_id in args["target_coin_ids"]]
-    largest = bool(args["largest"])
-    final_fee = uint64(int(Decimal(args["fee"]) * units["chia"]))
-    if number_of_coins > 500:
-        raise ValueError(f"{number_of_coins} coins is greater then the maximum limit of 500 coins.")
-    try:
-        wallet_type = await get_wallet_type(wallet_id=wallet_id, wallet_client=wallet_client)
-        mojo_per_unit = get_mojo_per_unit(wallet_type)
-    except LookupError:
-        print(f"Wallet id: {wallet_id} not found.")
-        return
-    if not await wallet_client.get_synced():
-        print("Wallet not synced. Please wait.")
-        return
-    is_xch: bool = wallet_type == WalletType.STANDARD_WALLET  # this lets us know if we are directly combining Chia
-    final_max_amount = uint64(int(max_amount * mojo_per_unit)) if not target_coin_ids else uint64(0)
-    final_min_coin_amount: uint64 = uint64(int(min_coin_amount * mojo_per_unit))
-    final_excluded_amounts: List[uint64] = [uint64(int(Decimal(amount) * mojo_per_unit)) for amount in excluded_amounts]
-    final_target_coin_amount = uint64(int(target_coin_amount * mojo_per_unit))
-    if final_target_coin_amount != 0:  # if we have a set target, just use standard coin selection.
-        removals: List[Coin] = await wallet_client.select_coins(
-            amount=(final_target_coin_amount + final_fee) if is_xch else final_target_coin_amount,
-            wallet_id=wallet_id,
-            max_coin_amount=final_max_amount,
-            min_coin_amount=final_min_coin_amount,
-            excluded_amounts=final_excluded_amounts + [final_target_coin_amount],  # dont reuse coins of same amount.
-        )
-    else:
-        conf_coins, _, _ = await wallet_client.get_spendable_coins(
-            wallet_id=wallet_id,
-            max_coin_amount=final_max_amount,
-            min_coin_amount=final_min_coin_amount,
-            excluded_amounts=final_excluded_amounts,
-        )
-        if len(target_coin_ids) > 0:
-            conf_coins = [cr for cr in conf_coins if cr.name in target_coin_ids]
-        if len(conf_coins) == 0:
-            print("No coins to combine.")
+async def async_combine(
+    *,
+    wallet_rpc_port: Optional[int],
+    fingerprint: Optional[int],
+    wallet_id: int,
+    fee: Decimal,
+    max_coin_amount: Decimal,
+    min_coin_amount: Decimal,
+    excluded_amounts: list[str],
```

Comment:
I don't have a strong opinion on this, but the types in this call seem geared towards passing in strings that are then converted in this function itself. e.g. this could be `List[Decimal]` (btw, why is it lower case `list` right now?).

and `target_coin_ids_str` could be `List[Bytes32]`. In a way, it would seem appropriate to perform the conversion from string -> true type in the caller.

Reply:
@arvidn, do we want to enforce the `str` -> `Decimal` conversion location in this PR?

Reply:
We'll take this up later.

Reply:
I think it would be fine to punt that to a second PR. @jack60612 would you mind doing that?